libs/Helpers/src/Helpers/QObjectVectorModel.h

```diff
@@ -0,0 +1,86 @@
+#pragma once
+#include <QAbstractListModel>
+#include <QQmlEngine>
+
+namespace Status::Helpers {
+
+/// Generic typed QObject provider model
+///
+/// Supports: source model update
+/// \todo rename it to SharedQObjectVectorModel
+template<typename T>
+class QObjectVectorModel final : public QAbstractListModel
+{
+    static_assert(std::is_base_of<QObject, T>::value, "Template parameter (T) not a QObject");
+
+public:
+
+    using ObjectContainer = std::vector<std::shared_ptr<T>>;
+
+    explicit QObjectVectorModel(ObjectContainer initialObjects, const char* objectRoleName, QObject* parent = nullptr)
```

Comment:
I would just settle for "modelData" for the role name, as it's used throughout QML models if there's one role only

Reply:
> I would just settle for "modelData" for the role name ...

This is still possible. Here I provide the option to add more context to the object exposed (`account`, `derived address` ...).

What do you think of the following exposing data model options, in the order of my preferences, by the provided context awareness:

```qml
// This.is the current case where `WalletAccount` is explicitly exposed as a role
DelegateExample {
  required property WalletAccount account
  Label {
    text: account.name
  }
}
```

vs.

```qml
// Here, we lose a little bit of context validation, and there is still a chance of having the wrong model set without any warning
// Use the `modelData` without explicitly exposing the role relying on the developer's awareness not to mix models but still providing a type check for QML
DelegateExample {
  property WalletAccount account: modelData
  Label {
    text: account.name
  }
}
```
vs.
```qml
// Same as above, but saves a line of code with the higher chance to have another model mixed up (`ChatAccount`, which also has a name)
DelegateExample {
  Label {
    text: modelData.name
  }
}
```

Reply:
The first option with `required property` is obviously the best :)